raiden/tests/unit/transfer/mediated_transfer/test_mediatorstate.py

```diff
@@ -0,0 +1,1447 @@
+        payment_network_identifier,
+        available_routes,
+        from_route,
+        from_transfer,
+    )
+
+    mediator_state = None
+    iteration = mediator.state_transition2(
+        mediator_state,
+        init_state_change,
+        channelmap,
+        block_number,
+    )
+
+    assert isinstance(iteration.new_state, MediatorTransferState)
+    assert iteration.new_state.transfers_pair[0].payer_transfer == from_transfer
+
+    msg = 'we have a valid route, the mediated transfer event must be emited'
+    assert iteration.events, msg
+
+    mediated_transfers = [e for e in iteration.events if isinstance(e, SendMediatedTransfer2)]
+    assert len(mediated_transfers) == 1, 'mediated_transfer should /not/ split the transfer'
+    send_transfer = mediated_transfers[0]
+    mediated_transfer = send_transfer.transfer
+
+    assert mediated_transfer.token == from_transfer.token, 'transfer token address mismatch'
+    assert mediated_transfer.lock.amount == from_transfer.lock.amount, 'transfer amount mismatch'
+    msg = 'transfer expiration mismatch'
+    assert mediated_transfer.lock.expiration < from_transfer.lock.expiration, msg
+    assert mediated_transfer.lock.hashlock == from_transfer.lock.hashlock, 'wrong hashlock'
+
+
+def test_no_valid_routes():
+    amount = UNIT_TRANSFER_AMOUNT
+    target = HOP2
+    from_expiration = HOP1_TIMEOUT
+
+    from_channel = factories.make_channel(
+        our_balance=amount,
+        partner_balance=amount,
+        partner_address=UNIT_TRANSFER_SENDER,
+        token_address=UNIT_TOKEN_ADDRESS,
+    )
+    from_route = factories.route_from_channel(from_channel)
+
+    from_transfer = factories.make_signed_transfer_for(
+        from_channel,
+        amount,
+        HOP1,
+        target,
+        from_expiration,
+        UNIT_SECRET,
+    )
+
+    channel1 = factories.make_channel(
+        our_balance=amount - 1,
+        token_address=UNIT_TOKEN_ADDRESS,
+    )
+    channel2 = factories.make_channel(
+        our_balance=0,
+        token_address=UNIT_TOKEN_ADDRESS,
+    )
+    available_routes = [
+        factories.route_from_channel(channel1),
+        factories.route_from_channel(channel2),
+    ]
+    channelmap = {
+        from_channel.identifier: from_channel,
+        channel1.identifier: channel1,
+        channel2.identifier: channel2,
+    }
+
+    block_number = 1
+    payment_network_identifier = factories.make_address()
+    init_state_change = ActionInitMediator2(
+        payment_network_identifier,
+        available_routes,
+        from_route,
+        from_transfer,
+    )
+
+    mediator_state = None
+    iteration = mediator.state_transition2(
+        mediator_state,
+        init_state_change,
+        channelmap,
+        block_number,
+    )
+    assert iteration.new_state is None
+
+    assert len(iteration.events) == 1
+    assert isinstance(iteration.events[0], SendRefundTransfer2)
+
+
+@pytest.mark.xfail(reason='Not implemented')
```

Comment:
Is this also issue 382? Does not seem to be from the description. More ilke a test for the proper lock time propagation so that we can't be cheated.

Regardless can you maybe also make an issue just to document we have a test that needs to be implemented?